extensions/guacamole-auth-wol/src/main/java/org/apache/guacamole/auth/wol/rest/WOLUserContextResource.java

```diff
@@ -0,0 +1,127 @@
+import org.apache.guacamole.net.auth.Directory;
+
+/**
+ * REST API endpoints for the Wake-on-LAN module
+ */
+@Produces(MediaType.APPLICATION_JSON)
+public class WOLUserContextResource {
+
+    /**
+     * The directory associated with this REST endpoint resource that
+     * will be queried to trigger WOL packets.
+     */
+    private final Directory<Connection> directory;
+    
+    /**
+     * Default timeout for attempting to connect to a host, in milliseconds.
+     */
+    public static final int DEFAULT_TIMEOUT = 5000;
+
+    /**
+     * Set up a new REST resource that will be use to trigger
+     * WOL packets.
+     * 
+     * @param directory
+     *     The directory to use for this REST endpoint.
+     */
+    public WOLUserContextResource(Directory<Connection> directory) {
+        this.directory = directory;
+    }
+
+    /**
+     * A method which takes a connection identifier and triggers the wake-up
+     * of the host, returning the connection identifier and true if the wake-up
+     * was sent successfully, and throwing an exception if an error occurs.
+     * 
+     * @param connectionIdentifier
+     *     The identifier of the connection to trigger the wake-up for.
+     * 
+     * @return
+     *     A map containing the connection identifier and a Boolean true if 
+     *     the wake-up packet was triggered successfully.
+     * 
+     * @throws GuacamoleException
+     *     If an error occurs triggering the host wake-up.
+     */
+    @POST
+    @Path("wake")
+    public Map<String, Boolean> wakeUpHost(
+            @FormParam("connectionIdentifier") String connectionIdentifier)
+            throws GuacamoleException {
+
+        // Retrieve the connection
+        WOLConnection connection = (WOLConnection) directory.get(connectionIdentifier);
+
+        // Send the wake-up packet
+        connection.wakeUpHost();
+
+        // If we don't get an exception, we assume the packet is sent.
+        return Collections.<String, Boolean>singletonMap(connectionIdentifier,
+                true);
+    }
+    
+    /**
+     * Have Java attempting to connect to the given hostname on the given
+     * port number, returning true if the connection is successful, otherwise
+     * false.
+     * 
+     * @param hostname
+     *     The hostname or IP of the system to connect to.
+     * 
+     * @param port
+     *     The port number on which to connect to the system.
+     * 
+     * @return
+     *     True if the connection is established successfully, otherwise false.
+     */
+    @GET
+    @Path("check/{hostname}/{port}")
+    public Boolean checkHost(@PathParam("hostname") String hostname,
+            @PathParam("port") int port) {
+        
+        try (Socket socket = new Socket()) {
+            socket.connect(new InetSocketAddress(hostname, port), DEFAULT_TIMEOUT);
+            return true;
+        }
+        catch (IOException | IllegalArgumentException | SecurityException e) {
+            return false;
```

Comment:
Any need to log that exception? Perhaps at least at the debug level?

Reply:
Yeah, that's reasonable.